### ISSUE TITLE
SB 5.2 Typescript build fail - (Addon-knobs)

### ISSUE THREAD

**Issue by sami616** [storybookjs/storybook#8105]
I'm running into a couple of issues when updating to 5.2.

Both errors seem to be caused by the addon knobs package.

1. Complains about a missing dep (manually installing  @types/react-color fixes this one)

```
ERROR in [at-loader] ./node_modules/@storybook/addon-knobs/dist/components/types/Color.d.ts:3:29 
    TS7016: Could not find a declaration file for module 'react-color'. '/Users/sami/Apps/components/node_modules/react-color/lib/index.js' implicitly has an 'any' type.
  Try `npm install @types/react-color` if it exists or add a new declaration (.d.ts) file containing `declare module 'react-color';`
```

2. Second complains about a missing export

```
ERROR in [at-loader] ./node_modules/@types/storybook__addon-knobs/index.d.ts:13:10 
    TS2305: Module '"../../../../../../../Users/sami/Apps/components/node_modules/@storybook/react/dist/client"' has no exported member 'RenderFunction'.
```

**Comment by pachtymi** [comment]
I believe that fix only fixes 1) but doesn't fix 2)

**Comment by hipstersmoothie** [comment]
>     TS2305: Module '"../../../../../../../Users/sami/Apps/components/node_modules/@storybook/react/dist/client"' has no exported member 'RenderFunction'.

This is because @sami616 is using `@types/storybook__addon-knobs`. Those types are now included with the package

**Comment by hipstersmoothie** [comment]
They can delete the dep on those types now.

**Comment by pachtymi** [comment]
ah, makes sense. 👍

**Comment by sami616** [comment]
Can confirm the `RenderFunction` error has gone now that i have uninstalled `@types/storybook__addon-knobs` - still seeing the `react-color` one though!

Am i right in thinking `@types/storybook__react` is now also not needed?

**Comment by tatsushitoji** [comment]
> @types/storybook__react is now also not needed?

i think so.
but still seeing `webpack-env` when i have uninstall `@types/storybook__react`.

**Comment by shilman** [comment]
Yippee!! I just released https://github.com/storybookjs/storybook/releases/tag/v5.3.0-alpha.1 containing PR #8118 that references this issue. Upgrade today to try it out!

You can find this prerelease on the `@next` NPM tag.

**Comment by sami616** [comment]
Thanks @shilman just upgraded and tried running my app, and the build is failing with the following error:

```
Module not found: Error: Can't resolve '@storybook/core/client' in '/Users/sami/Apps/components/node_modules/@storybook/react/dist/client/preview'
```

I could probably just manually install the dep to resolve but thought its worth pointing out anyway!

**Comment by jamie--stewart** [comment]
Hi, did you get a chance to release that fix? ☝️

No worries if not, happy to work-around for just now. Was _very_ happy to find this issue!

**Comment by sami616** [comment]
`5.3.0-alpha.2` looks like its on the `@next` tag it resolved the issue for me

**Comment by ndelangen** [comment]
@shilman can we release a patch containing this?

**Comment by SpainTrain** [comment]
This issue still exists in 5.3 alpha 6 with `@storybook/addon-info`.  Should this be a separate issue? Thanks!

**Comment by jacobrask** [comment]
If anyone gets blocked by this, I can recommend adding `skipLibCheck: true` to your `tsconfig.json` - because the internal typings of Storybook don't really matter that much to the type safety of your application code.

**Comment by shilman** [comment]
Yee-haw!! I just released https://github.com/storybookjs/storybook/releases/tag/v5.2.2 containing PR #8118 that references this issue. Upgrade today to try it out!
